File: app/src/main/java/com/bopr/android/smailer/sync/AppContentProvider.java
package com.bopr.android.smailer.sync;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class AppContentProvider extends ContentProvider {

    public static final String AUTHORITY = "com.bopr.android.smailer";
    public static final Uri CONTENT = Uri.parse("content://" + AUTHORITY + "/");
//    public static final Uri CONTENT_EVENTS = Uri.withAppendedPath(CONTENT, "events");

    @Override
    public boolean onCreate() {
        return true;
    }

    @Nullable
    @Override
    public String getType(@NonNull Uri uri) {
        return null;
    }

    protected boolean isValidUri(@NonNull Uri uri) {
        return AUTHORITY.equals(uri.getAuthority());
    }

    @Nullable
    @Override
    public Cursor query(@NonNull Uri uri, @Nullable String[] projection, @Nullable String selection,
                        @Nullable String[] selectionArgs, @Nullable String sortOrder) {
//        if (isValidUri(uri)) {
//            return database.query(uri.getLastPathSegment(), projection, selection, selectionArgs, sortOrder);
//        }
        return null;
    }

    @Nullable
    @Override
    public Uri insert(@NonNull Uri uri, @Nullable ContentValues values) {
//        if (isValidUri(uri)) {
//            database.put(uri.getLastPathSegment(), values);
//            return uri;
//        }
        return null;
    }

    @Override
    public int delete(@NonNull Uri uri, @Nullable String selection, @Nullable String[] selectionArgs) {
//        if (isValidUri(uri)) {
//            return database.delete(uri.getLastPathSegment(), selection, selectionArgs);
//        }
        return 0;
    }

    @Override
    public int update(@NonNull Uri uri, @Nullable ContentValues values, @Nullable String selection,
                      @Nullable String[] selectionArgs) {
//        if (isValidUri(uri)) {
//            return database.update(uri.getLastPathSegment(), values, selection, selectionArgs);
//        }
        return 0;
    }
}

File: app/src/main/kotlin/com/bopr/android/smailer/sync/AppContentProvider.kt
package com.bopr.android.smailer.sync

import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri

/**
 * Application content provider. Does nothing.
 *
 * Required by synchronization framework.
 * Without it sync adapter won't call [SyncAdapter.onPerformSync]
 *
 * @author Boris Pronin ([boprsoft.dev@gmail.com](mailto:boprsoft.dev@gmail.com))
 */
class AppContentProvider : ContentProvider() {

    override fun onCreate(): Boolean {
        return true
    }

    override fun getType(uri: Uri): String? {
        return null
    }

    override fun query(uri: Uri, projection: Array<String>?, selection: String?,
                       selectionArgs: Array<String>?, sortOrder: String?): Cursor? {
        return null
    }

    override fun insert(uri: Uri, values: ContentValues?): Uri? {
        return null
    }

    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<String>?): Int {
        return 0
    }

    override fun update(uri: Uri, values: ContentValues?, selection: String?,
                        selectionArgs: Array<String>?): Int {
        return 0
    }

    companion object {

        const val AUTHORITY = "com.bopr.android.smailer"
    }
}